File: app/src/main/java/com/example/spacetraders/data/views/RefuelActivity.java
package com.example.spacetraders.data.views;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.example.spacetraders.R;
import com.example.spacetraders.data.models.Interactor;
import com.example.spacetraders.data.entity.Character;


public class RefuelActivity extends AppCompatActivity {

    private Character character;
    private TextView shipType;
    private TextView cargoSize;
    private TextView health;
    private TextView weaponsSize;
    private TextView fuelEfficiency;
    private TextView basePrice;
    private TextView fuelLeft;
    private TextView currency;
    private Button refuelButton;
    private Button upgradeButton;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_refuel);

        character = Interactor.getInteractor().getCharacter();


        shipType = findViewById(R.id.displayType);
        cargoSize = findViewById(R.id.displayCargoSize);
        health = findViewById(R.id.displayCurrHealth);
        weaponsSize = findViewById(R.id.displayWeaponsNum);
        fuelEfficiency = findViewById(R.id.displayFuelEfficiency);
        basePrice = findViewById(R.id.displayBasePrice);
        fuelLeft = findViewById(R.id.displayFuelLeft);
        currency = findViewById(R.id.displayCurrency);
        refuelButton = findViewById(R.id.buttonRefuel);
        upgradeButton = findViewById(R.id.buttonUpgrade);

        shipType.setText(character.getShip().getName());
        cargoSize.setText(""+ character.getShip().getCargoSize());
        health.setText("" + character.getShip().getCurrentHealth());
        weaponsSize.setText("" + character.getShip().getMaxWeaponsAmount());
        fuelEfficiency.setText(""+ character.getShip().getFuelEfficiency());
        basePrice.setText(""+ character.getShip().getBasePrice());
        fuelLeft.setText(""+ character.getShip().getFuel());
        currency.setText("" + character.getCurrency() + "c");

        //int amountForRefuel = character.getShip().getFuelEfficiency() - character.getShip().getFuel();
        //1 fuel =  1 credit
        refuelButton.setText("1 Fuel = 5c");
        refuelButton.setOnClickListener((View v) -> {
           if(character.getShip().getFuel() == character.getShip().getFuelEfficiency()) {
               Toast.makeText(getApplicationContext(), "Tank Full", Toast.LENGTH_LONG).show();
               return;
           } else if (character.getCurrency() < 5) {
               Toast.makeText(getApplicationContext(), "Insufficient", Toast.LENGTH_LONG).show();
               return;
           } else {
               character.setCurrency(character.getCurrency() - 5);
               currency.setText("" + character.getCurrency() + "c");
               character.getShip().setFuel(character.getShip().getFuel() + 1);
               fuelLeft.setText("" + character.getShip().getFuel());
           }
        });

        upgradeButton.setOnClickListener((View v) -> {
            Intent intent = new Intent(RefuelActivity.this, ShipYardActivity.class);
            startActivity(intent);
        });


    }
}

File: app/src/main/java/com/example/spacetraders/data/views/RefuelActivity.kt
package com.example.spacetraders.data.views

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

import com.example.spacetraders.R
import com.example.spacetraders.data.models.Interactor
import com.example.spacetraders.data.entity.Character


class RefuelActivity : AppCompatActivity() {

    private var character: Character? = null
    private var shipType: TextView? = null
    private var cargoSize: TextView? = null
    private var health: TextView? = null
    private var weaponsSize: TextView? = null
    private var fuelEfficiency: TextView? = null
    private var basePrice: TextView? = null
    private var fuelLeft: TextView? = null
    private var currency: TextView? = null
    private var refuelButton: Button? = null
    private var upgradeButton: Button? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_refuel)

        character = Interactor.interactor.character


        shipType = findViewById(R.id.displayType)
        cargoSize = findViewById(R.id.displayCargoSize)
        health = findViewById(R.id.displayCurrHealth)
        weaponsSize = findViewById(R.id.displayWeaponsNum)
        fuelEfficiency = findViewById(R.id.displayFuelEfficiency)
        basePrice = findViewById(R.id.displayBasePrice)
        fuelLeft = findViewById(R.id.displayFuelLeft)
        currency = findViewById(R.id.displayCurrency)
        refuelButton = findViewById(R.id.buttonRefuel)
        upgradeButton = findViewById(R.id.buttonUpgrade)

        shipType!!.text = character!!.ship!!.name
        cargoSize!!.text = "" + character!!.ship!!.cargoSize
        health!!.text = "" + character!!.ship!!.getCurrentHealth()
        weaponsSize!!.text = "" + character!!.ship!!.maxWeaponsAmount
        fuelEfficiency!!.text = "" + character!!.ship!!.fuelEfficiency
        basePrice!!.text = "" + character!!.ship!!.basePrice
        fuelLeft!!.text = "" + character!!.ship!!.getFuel()
        currency!!.text = "" + character!!.currency + "c"

        //int amountForRefuel = character.getShip().getFuelEfficiency() - character.getShip().getFuel();
        //1 fuel =  1 credit
        refuelButton!!.text = "1 Fuel = 5c"
        refuelButton!!.setOnClickListener { v: View ->
            if (character!!.ship!!.getFuel() == character!!.ship!!.maxFuel) {
                Toast.makeText(applicationContext, "Tank Full", Toast.LENGTH_LONG).show()
            } else if (character!!.currency < 5) {
                Toast.makeText(applicationContext, "Insufficient", Toast.LENGTH_LONG).show()
            } else {
                character!!.currency = character!!.currency - 5
                currency!!.text = "" + character!!.currency + "c"
                character!!.ship!!.setFuel(character!!.ship!!.getFuel() + 1)
                fuelLeft!!.text = "" + character!!.ship!!.getFuel()
            }
        }

        upgradeButton!!.setOnClickListener { v: View ->
            val intent = Intent(this@RefuelActivity, ShipYardActivity::class.java)
            startActivity(intent)
        }


    }
}
